core/src/main/java/org/eclipse/hono/util/TenantObject.java

```diff
@@ -505,9 +507,56 @@ public static JsonObject newAdapterConfig(final String type, final boolean enabl
+        return Optional.ofNullable(getProperty(TenantConstants.FIELD_RESOURCE_LIMITS))
+                .map(limits -> getProperty((JsonObject) limits, TenantConstants.FIELD_MAX_CONNECTIONS,
                         (Number) TenantConstants.DEFAULT_MAX_CONNECTIONS).longValue())
                 .orElse(TenantConstants.DEFAULT_MAX_CONNECTIONS);
     }
+
+    /**
+     * Gets the maximum number of bytes to be allowed for the time period defined by the
+     * {@link TenantObject#getPeriodInDays()}.
+     *
+     * @return The maximum number of bytes or {@link TenantConstants#DEFAULT_MAX_BYTES} if not set.
+     */
+    @JsonIgnore
+    public long getMaximumNumberOfBytes() {
+        return Optional.ofNullable(getProperty(TenantConstants.FIELD_RESOURCE_LIMITS))
+                .map(limits -> getProperty((JsonObject) limits, TenantConstants.FIELD_DATA_VOLUME))
+                .map(dataVolumeLimits -> getProperty((JsonObject) dataVolumeLimits, TenantConstants.FIELD_MAX_BYTES,
+                        (Number) TenantConstants.DEFAULT_MAX_BYTES).longValue())
+                .orElse(TenantConstants.DEFAULT_MAX_BYTES);
+    }
+
+    /**
+     * Gets the number of days for which the data usage is calculated and compared against the
+     * {@link TenantObject#getMaximumNumberOfBytes()}.
+     *
+     * @return The number of days for which the data usage is calculated or
+     *         {@link TenantConstants#DEFAULT_PERIOD_IN_DAYS} if not set.
+     */
+    public long getPeriodInDays() {
+        return Optional.ofNullable(getProperty(TenantConstants.FIELD_RESOURCE_LIMITS))
+                .map(limits -> getProperty((JsonObject) limits, TenantConstants.FIELD_DATA_VOLUME))
+                .map(dataVolumeLimits -> getProperty((JsonObject) dataVolumeLimits, TenantConstants.FIELD_PERIOD_IN_DAYS,
+                        (Number) TenantConstants.DEFAULT_PERIOD_IN_DAYS).longValue())
+                .orElse(TenantConstants.DEFAULT_PERIOD_IN_DAYS);
+    }
+
+    /**
+     * Gets the date on which the data volume limit came into effect.
+     * <p>
+     * The date string should comply to the {@link DateTimeFormatter#ISO_LOCAL_DATE}
+     *
+     * @return The date on which the data volume limit came into effective or {@code null} if not set.
+     * @throws DateTimeParseException if the date string fails to comply with the
+     *             {@link DateTimeFormatter#ISO_LOCAL_DATE} format.
+     */
+    public LocalDate getEffectiveSince() {
+        return Optional.ofNullable(getProperty(TenantConstants.FIELD_RESOURCE_LIMITS))
+                .map(limits -> getProperty((JsonObject) limits, TenantConstants.FIELD_DATA_VOLUME))
+                .map(dataVolumeLimits -> getProperty((JsonObject) dataVolumeLimits, TenantConstants.FIELD_EFFECTIVE_SINCE))
+                .map(effectiveSince -> LocalDate.parse((String) effectiveSince, DateTimeFormatter.ISO_LOCAL_DATE))
+                .orElse(null);
+    }
```

Comment:
I am not sure if this is helpful. A client needs to retrieve all these properties individually and then pass them into the constructor of a `ResourceLimitChecks` instance. Wouldn't it make more sense to just use a *getResourceLimits* method to retrieve the `resource-limits* JSON object and then pass that in to the constructor?

Reply:
These properties are not passed into the constructor of a `ResourceLimitChecks` instance. `ResourceLimitChecks` methods take a `TenantObject` instance as an argument and retrieve the individual properties.  Its makes more sense to have one _getResourceLimits_ method in `TenantObject`, so that any custom implementation of `ResourceLimitChecks` need not worry about the `TenantObject` class.

I will add one method _getResourceLimits_ to the `TenantObject ` and move the above methods to the default `ResourceLimitChecks` implementation which is `PrometheusBasedResourceLimitChecks`.

Reply:
yes, I think that would be better, do you agree?

Reply:
yes. Also there is no need to define properties names _(starting with FIELD\_*)_ and default values in `TenantConstants`.

Reply:
As we discussed, I have made changes and pushed it.